README.md

```diff
@@ -207,23 +220,265 @@ More information can be found [here](http://www.diveintopython3.net/strings.html
+
+## Examples:
+
+In this example, I will demonstrate how we generate `synapser` and `synapserutils` packages by wrapping Python package `synapsePythonClient`.
+
+`synapsePythonClient` has the following structure:
+```
+synapsePythonClient package
+    module: synapseclient
+        module: client
+            class: Synapse
+                function: get
+                function: login
+        module: entity
+            class: Entity
+                function: get
+                function: privateGet
+            class: File
+                function: get
+                function: privateGet
+            class: Folder
+                function: get
+                function: privateGet
+        module: table
+            function: Table
+            function: column_ids
+            class: Column
+            class: CsvFileTable
+        module: cache
+            function: set
+    module: synapseutils
+        module: copy
+            function: copy
+            function: copyWiki
+            function: copyFileHandles
+        module: sync
+            function: syncToSynapse
+            function: syncFromSynapse
+        module: monitor
+            function: notifyMe
+```
+
+### Expose all functions and classes within a Python module
+
+For the R package `synapserutils`, our first attempt is to exposing all functions under `synapseutils` Python module. In `.configure` file, we execute an R script that calls `generateRdFiles` as following:
+```r
+generateRdFiles(srcRootDir,
+                pyPkg = "synapseutils",
+                module = "synapseutils")
+```
+
+Where `srcRootDir` is the path to `synapserutils` directory, `pyPkg` is the name of the Python package, and `module` is the name of the Python module. In this case, `pyPkg` and `module` has the same name. In later examples, we will use different names for `pyPkg` and `module`.
+
+Then in `.onLoad` under `synapserutils/R/zzz.R`, we add the following:
+```r
+callback <- function(name, def) {
+  setGeneric(name, def)
+}
+generateRWrappers(pyPkg = "synapseutils",
+                  module = "synapseutils",
+                  setGenericCallback = callback)
+```
+For the R wrappers to be available in `synapserutils` package namespace, `setGeneric` must be defined in the `synapserutils` package. Therefore, we need to define it in `synapserutils` and pass it through `generateRWrappers`.
+
+For more information about how to use `setGeneric`, please view its reference documentation by:
+```r
+?setGeneric
+```
+
+### Expose a subset of functions within a Python module
+
+For many reasons, some Python functions are only meaningful to Python users. We would not want to expose those functions in our R package.
```

Comment:
some Python functions are meant to be used internally, and these do not need to be exposed the R package.

Reply:
for example, Python decorators